### PR TITLE
[v13] Allow for Windows PKI operations to target a different domain

### PR DESCRIPTION
**Pull request by zmb3**
Today, our AD support largely assumes there is a single active directory domain. The certificates that we generate are for users in this domain, the computers we discover via LDAP come from this domain, and the PKI set up we perform targets this domain.

In more complicated AD configurations, PKI is often configured in a root domain, while users, servers, and discovery should be done against a child domain.

The new pki_domain configuration field will allow you to override the default domain specified in the ldap section with a root domain that is used for configuring the NTAuth store and publishing the CRL. Teleport continues to do discovery and issue certificates for the domain specified in the ldap section of the config.

Updates #20233

**Note:** targeting v13 to do a dev build

### CODE REVIEW

File: lib/config/fileconf.go [hunk 1]
```diff
@@ -2377,6 +2377,12 @@ type WindowsDesktopService struct {
 	ShowDesktopWallpaper bool `yaml:"show_desktop_wallpaper,omitempty"`
 	// LDAP is the LDAP connection parameters.
 	LDAP LDAPConfig `yaml:"ldap"`
+	// PKIDomain optionally configures a separate Active Directory domain
+	// for PKI operations. If empty, the domain from the LDAP config is used.
+	// This can be useful for cases where PKI is configured in a root domain
+	// but Teleport is used to provide access to users and computers in a child
+	// domain.
+	PKIDomain string `yaml:"pki_domain"`
```
Comment (ibeckermayer) [inline lib/config/fileconf.go:2385]: Maybe it should be a subfield of the `ldap` object?

File: lib/srv/desktop/windows_server.go [hunk 2]
```diff
@@ -352,15 +358,20 @@ func NewWindowsService(cfg WindowsServiceConfig) (*WindowsService, error) {
 		auditCache:  newSharedDirectoryAuditCache(),
 	}
 
+	caLDAPconfig := s.cfg.LDAPConfig
+	if s.cfg.PKIDomain != "" {
+		s.cfg.Log.Infof("Windows PKI will be performed against %v", s.cfg.PKIDomain)
+		caLDAPconfig.Domain = s.cfg.PKIDomain
+	}
```
Comment (ibeckermayer) [inline lib/srv/desktop/windows_server.go:365]:
```suggestion
	if s.cfg.PKIDomain != "" {
		caLDAPconfig.Domain = s.cfg.PKIDomain
	}
	s.cfg.Log.Infof("Windows PKI will be performed against %v", caLDAPconfig.Domain)
```

File: lib/config/fileconf.go (hunk 1 above)
Comment (zmb3) [reply lib/config/fileconf.go:2385]: I don't think so. The `ldap` object should be more about "how do I connect to an LDAP server" as per

```
// LDAPConfig is the LDAP connection parameters.
```

Putting the domain and SID in there was a mistake IMO.

**Comment by zmb3** [comment]
Opened #33218 against master now that we've proven this functionality out.

**Pull request closed without merging**